### DISCUSSION TITLE
Getting `resolver redeclared in this block`

### DISCUSSION THREAD

**Discussion by tlenexkoyotl** [99designs/gqlgen#1821] [Q&A]
### What happened?
I am building a GraphQL server to test gqlgen, all the code is in this public repo https://gitlab.com/tlenexkoyotl/gql-golang-demo/ and it includes instructions for every step I've taken through this test.

I separated my `schema.resolvers` into separate files (`mutationResolver.go`, `queryResolver.go`, `userResolver.go`, `meetupResolver.go`) as you can see, since a tutorial I'm following does so as a *good practice*.

But regenerating gqlgen's boilerplate code with `go run github.com/99designs/gqlgen` results in a bunch of `resolver redeclared in this block` errors.

### What did you expect?

In the tutorial, whenever `go run github.com/99designs/gqlgen` is ran again to regenerate boilerplate code, it does not copy all implemented resolvers back into the `resolver.go` file, only generates new code. I would expect the same to happen in my project.

### Minimal graphql.schema and models to reproduce
```graphql
type User {
  id: ID!
  username: String!
  email: String!
  
  meetups: [Meetup!]!
}

type Meetup {
  id: ID!
  name: String!
  description: String!
  user: User!
}

type Query {
  meetups: [Meetup!]!
}
```

### versions
 - `gqlgen version`?  I have tried `v0.12.2` and `v0.13.0`, both present the same issue (the tutorial used `v0.10.2`)
 - `go version`? Go `1.15`
 - dep or go modules? Go modules

**Comment by HectorMalot** [comment]
_Edit: I solved it for my situation. Leaving this if you're stuck in the same situation:_ 

Note that with `layout: follow-schema` in your config, `gqlgen` will ONLY look at the expected resolvers file when generating the resolvers. Other files are not taken into account and so this could lead to generation of types and functions with identical signatures.

So, given the following schema definitions:
```
./users.graphqls
./schema.graphqls
```

gqlgen will generate 2 files with resolver stubs: `users.resolvers.go` and `schema.resolvers.go`. It will also only check in these files for existing resolvers. Existing resolvers in other files are not taken into account.

Depending on how you look at this, it might be considered either a bug (if you read the config as "generate my resolvers as follows") or desired behaviour (if you read the config as "all resolvers _must_ be in these files).

---
# Old comment

I am facing the same issue. Even if I copy the generated resolvers directly into a new user_resolver.go, it regenerates the stub resolvers on the next try and gives me:

```
/user_resolver.go:12:6: 	other declaration of userResolver
/schema.resolvers.go:137:20:    method User already declared for type Resolver struct{}
/user_resolver.go:15:20: 	other declaration of User
```

relevant gqlgen configuration:
```
resolver:
  layout: follow-schema
  dir: pkg/server/resolvers
  package: resolvers
```

gqlgen version: `v0.13.0`
go version: `1.16.2`
go modules: `yes`

**Comment by Igorpollo** [comment]
@HectorMalot how about the queries and mutations? Did you found a solution?
My queries and mutations still get copied to schema.resolvers

**Comment by HectorMalot** [comment]
@Igorpollo, if you've defined them in `schema.graphqls`, then they will be created in `schema.resolvers.go`, if you want them in a separate file, I suppose you could define e.g. queries in `queries.graphqls` and they would end up in `queries.resolvers.go`. Or does that also not work for you?

If I find some time, I might open a PR to improve documentation on the `layout: follow-schema` option, as this behaviour was quite counterintuitive for me. (Alternatively, adopt the behaviour that if the resolver is defined _anywhere_ in the package, that it won't be generated.
